chain/starknet/src/chain.rs

```diff
@@ -391,20 +392,61 @@ impl TriggersAdapterTrait<Chain> for TriggersAdapter {
                 transaction
                     .events
                     .iter()
-                    .map(|event| {
-                        StarknetTrigger::Event(StarknetEventTrigger {
-                            event: Arc::new(event.clone()),
-                            block: shared_block.clone(),
-                            transaction: transaction.clone(),
-                        })
+                    .filter_map(|event| {
```

Comment:
As mentioned in a different comment, would be great if you could move the matching logic into the filter itself leave this implementation focused on extract the triggers from the block rather than dealing with internals from the filter

Reply:
Would also be great if we can have some tests, near has some examples you can follow.